### DISCUSSION TITLE
Album and folder 'Categories'

### DISCUSSION THREAD

**Discussion by martins250** [photoprism/photoprism#2010] [Q&A] [answered]
I have been giving Photoprism a full trial to discover its features. It is going to make a significant contribution to photo library software. One matter I have come across is to do with Album 'categories'. If you edit an album, you can assign a 'category' to it, which appears like  a 'tag' underneath any album to which a category has been assigned and a drop down list can appear in the search bar for Albums.  This is quite possibly a useful feature but it seems to be undocumented. I have several questions:

1. Is this a feature which will be supported in the future, or is it something which has been tested and forgotten about, and will later be removed?
2. Such categories can also be applied to 'Folders' in the same way. There appears to be one common list of categories which can be used to filter Albums or Folders. That is OK but of course users may find that selecting a category on the drop down list for Albums produces no results
3. On my Android, the heading 'All Categories' for Albums does not appear in portrait mode, but does in landscape.
4. There might be confusion over the term 'Categories' as the option 'All Categories' appears in the general search boxes (the last of the 8 boxes), but the options for selection here mean 'Labels'.

Thank you

**Comment by graciousgrey** [comment, marked as answer]
Hello,

thanks a lot for your feedback!

1) The categories for albums will stay. Some improvements are already planned: https://github.com/photoprism/photoprism/issues/578 as well. We just have other tasks with a higher priority for now.

2)  You can set a category for albums, folders, moments, states and calendar months (all these are types of albums). You are right, if you set the category "family" for an album it also appears on the category dropdown within folders. This is something we can improve in the future.

3) [We already have a ticket so improve this](https://github.com/photoprism/photoprism/issues/1896)

4) Categories in the photo search refer to categories of labels such as animal. Categories on the album searches refer to album categories.  We might rename it when we find time to improve this feature. Any ideas?

**Reply by martins250** [reply]
Thank you for your helpful response.

Following on from point 4, you may be able to explain the context of these 'categories of labels' which arise in the boxes of general search. 

As far as I can see, then these are what one might call 'general labels or super-labels', but I do not know where they come from or how they are chosen. In my test of some 2500 photos, then 63 ordinary labels are derived (aircraft, alpine, altar ... ). In the general 'Search' menu, then there are 23 'categories of labels' (airport, animal, architecture ...). Some of these categories are the same as labels (eg architecture, beach). In such cases, it seems the filter produces the same selection of photos. But some categories which appear (eg airport) are not in 'Labels', when you might expect them to. I am just bit but confused.

It looks as though, unlike 'Labels', these 23 'categories of labels' cannot be edited or added to. My question must be whether people ever need to classify labels, or if they do, it might be better if they could edit the categories. Or maybe there is another purpose which I am overlooking? (eg allowing an AND search involving categories of labels)

**Reply by graciousgrey** [reply]
The categories of labels cannot be edited. They are coming from the classification of our object recognition (see: https://github.com/photoprism/photoprism/blob/develop/internal/classify/rules.yml)

**Reply by martins250** [reply]
Thank you for that explanation.  I have also now come across the response in  #1672.

I think I can see what is happening now - there are labels which are very specific (eg tractor) and can be user-defined, and then there are the collective/generic labels (eg vehicle). I am not sure how many people will routinely use both types of labels. Neither am I sure who devised them and links between the two (eg why are 'beach' and 'sand' both generic/category labels?). But I guess some may find this distinction useful, even if the generic/category labels cannot be changed.

I have 3 comments:

1. The documentation could make this distinction clear. At the moment all I can find is "To see only labels with high priority click <the eye> in the upper right corner. On screen, hovering over <the eye> just says "Show More". As you might have intimated earlier, it would be good to find a word to replace 'Categories' in the Photos search boxes. ('Generic Labels' is the best I can think of, but it is not very snappy!)
2. The default choice from the main menu is to have a large number of labels, or alternatively you can add further labels if you opt to include the collective/generic labels. Maybe there could be a third category just to see only the collective/generic labels (eg "Show Fewer")? Alternatively, but more complex to incorporate, users could choose from the collective/generic labels first, and then a drop-down box of the associated and more specific labels would follow. 
3. Ideally, users would be able to easily amend or delete the collective/generic labels if they wished, but that could be a long-term idea.

**Reply by lastzero** [reply]
You are welcome to help improving our docs :)

There's some background information in our User Guide FAQs at https://docs.photoprism.app/user-guide/faq/:

> **What's the difference between keywords and labels?**
>
> Keywords contain a list of search terms extracted from metadata, file names, and other sources like geodata. Pictures with matching keywords automatically show up in related Labels.
> 
> Although related, keywords and labels serve different purposes:
> 
> - Labels may have parent categories and are primarily used for classification, like "animal", "cat", or "boat". Duplicates and ambiguities should be avoided.
>
> - Keywords are primarily used for searching. They may include similar terms and translations, like "kitten", "kitty", and "cat".

The complete classification rules can be found in our public source code repository (human-friendly YAML file):

https://github.com/photoprism/photoprism/blob/develop/internal/classify/rules.yml

It took us more than a year to configure each of the 1000 labels. We are aware that the result does not meet 100% of the needs of all users.  

However, we didn't feel that developing a user-friendly settings interface and writing extensive documentation (so users wouldn't break anything, or at least know why) would provide the most benefit at this point.

For example, multi-user features and also translating the existing labels seem much more important before we can even think about touching the labels/categories and possibly adding more settings. So for now, this is only possible for developers. Improving the docs would be a good start!

**Reply by martins250** [reply]
Thanks for that.  I will have a go at making a small edit to your docs.
